File: Model/Business/TeamManager.vb
Public Class TeamManager

    Public Sub New()
        Load()
    End Sub

    Private Teams(-1) As Team

    Public Function NewTeam() As Team
        Return New Team(Teams.Length + 1)
    End Function

    Public Sub AddTeam(team As Team)
        Dim index As Integer

        index = Teams.Length
        ReDim Preserve Teams(index)
        Teams(index) = team
    End Sub

    Public Sub RemoveTeam(teamToRemove As Team)
        Dim index As Integer

        For Each team In Teams
            If Not team.Equals(teamToRemove) Then
                Teams(index) = team
                index += 1
            End If
        Next team

        ReDim Preserve Teams(index - 1)
    End Sub

    Public Function GetTeams() As Team()
        Return Teams
    End Function

    Private Sub Load()
        Dim data As Object = Database.Load(GetType(Team()))
        If Not data Is Nothing Then Teams = CType(data, Team())
    End Sub

End Class

File: POFF.Kicker/Domain/TeamManager.cs
using POFF.Kicker.Domain;
using POFF.Kicker.Infrastructure;
using System;

namespace POFF.Kicker.Domain;

public class TeamManager
{

    public TeamManager()
    {
        Load();
    }

    private Team[] Teams = new Team[0];

    public Team NewTeam()
    {
        return new Team(Teams.Length + 1);
    }

    public void AddTeam(Team team)
    {
        int index;

        index = Teams.Length;
        Array.Resize(ref Teams, index + 1);
        Teams[index] = team;
    }

    public void RemoveTeam(Team teamToRemove)
    {
        var index = default(int);

        foreach (var team in Teams)
        {
            if (!team.Equals(teamToRemove))
            {
                Teams[index] = team;
                index += 1;
            }
        }

        Array.Resize(ref Teams, index);
    }

    public Team[] GetTeams()
    {
        return Teams;
    }

    private void Load()
    {
        var data = Database.Load(typeof(Team[]));
        if (data is not null)
            Teams = (Team[])data;
    }

}